### ISSUE TITLE
Reader: Posts with disabled likes can still receive likes via Reader

### ISSUE THREAD

**Issue by fresatomica** [Automattic/wp-calypso#56239]
### Quick summary

Even though you disable posts for your blog, users will still be able to like them via Reader:

Site view:
<img width="414" alt="Screen Shot on 2021-09-14 at 16-29-33" src="https://user-images.githubusercontent.com/40177971/133278078-90957749-cce6-438c-8fb3-2b3d5a6039d2.png">

Reader view:
<img width="231" alt="Screen Shot on 2021-09-14 at 16-31-14" src="https://user-images.githubusercontent.com/40177971/133278163-e3191e55-9a8d-499f-8a59-9ca542ccc07e.png">

### Steps to reproduce

1. Disable likes for your posts
2. Search for your post on the Reader
3. You'll still be able to like the post from there

### What you expected to happen

The likes should be disabled everywhere.

### What actually happened

You can like a post, even though its author disables likes on their blog.

### Context

A user reported this via support forums with this comment:

> I would like a feature that prevents people in the Reader from "liking" anything, when I have chosen to have "likes" disabled! Surely, as it's my blog, it should be my choice all round as to whether I want fake "likes" from the Reader?

### Operating System

macOS

### Reproducibility

Consistent

### Severity

All

**Comment by Robertght** [comment]
cc @Automattic/flow-patrol-create

**Comment by yansern** [comment]
When reader retrieves post content, the API URL `public-api.wordpress.com/rest/v1.2/read/feed` is called. One of the property returned in the JSON response is `likes_enabled`. And this value should be `false`.

![image](https://user-images.githubusercontent.com/1287077/134326583-0c45b340-e729-4ba4-96cc-5a46b42a629a.png)

The function to start looking at is `process_feedbag_item()` in `public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-read-v1-2-endpoint.php`. 

The site option to retrieve is `get_option( 'disabled_likes' ) );`. However, reader feeds can come from Simple Sites or Atomic.

For simple sites, retrieving the option is easy.
For atomic sites, retrieving the option seems to be coming from a "shadow blog"? See `add_wpcom_shadow_blog_data_to_jetpack_post()`.

What needs further verification is whether `disabled_likes` option between the actual atomic site and the shadow blog is on parity.

Another endpoint that might need to be looked at is `public-api.wordpress.com/rest/v1.2/read/following`. That's for the Reader page where a list of posts are shown. It seems that Like button is also shown there too.

**Comment by roo2** [comment]
When I retested this it appeared that the endpoint was returning correct results, but the front end just wasn't handling the data yet. (tested a simple site with likes disabled sitewide)

I've created a fix and will also check atomic sites and sites with likes on a single page disable

https://github.com/Automattic/wp-calypso/pull/61306

**Comment by roo2** [comment]
It looks like this is something that was done intentionally in order to give people a consistent reader experience. https://github.com/Automattic/wp-calypso/pull/61306#issuecomment-1046415814 we also don't currently have a designer or anyone assigned to the reader project who is making these decisions so I think we should close this as won't fix.

I'd explain this as the "show likes button" setting is only meant to control whether the button is visible on your pages, but the likes feature of wordpress.com and the reader, also exists for the benefit of readers, in order to save something and come back to it, so we don't give an option to authors to completely disable that feature.

Also, AFAIK at this time we don't suffer the same problems with many bots creating fake likes as platforms like twitter and Facebook, so if the user's getting likes from the reader, they're probably real people!

**Comment by zdenys** [comment]
> It looks like this is something that was done intentionally in order to give people a consistent reader experience. https://github.com/Automattic/wp-calypso/pull/61306#issuecomment-1046415814 

I encountered a case 6328131-zd-woothemes where the user wanted to disable the Like functionality and rightfully could not do that fully as liking remains possible via Reader. 

> we also don't currently have a designer or anyone assigned to the reader project who is making these decisions so I think we should close this as won't fix.

We now have @Automattic/loop 🥳 Thus I'll reopen this issue at least so it can be revisited. It can be closed again if no action will be taken. Let's just follow up with the user to let them know our decision. Thanks!

TL;DR: If a user disables Like buttons on specific/all Posts, this decision should be respected by Reader and the Like button hidden there as well.

**Comment by davemart-in** [comment]
I think @roo002 makes a good point about the desire for consistency in the reader and the ability for readers to use likes to bookmark posts they like. I suspect that if I can across a post in the reader without a like button I'd assume it was a bug (which we'd likely get more reports for).

For now, my vote would be to hold off on prioritizing this. Happy to re-evaluate if a number of other customers requests come in.
